src/vs/platform/terminal/node/windowsShellHelper.ts

```diff
@@ -55,35 +64,22 @@ export class WindowsShellHelper extends Disposable implements IWindowsShellHelpe
 		if (this._isDisposed) {
 			return;
 		}
-
-		// The debounce is necessary to prevent multiple processes from spawning when
-		// the enter key or output is spammed
-		Event.debounce(this._onCheckShell.event, (l, e) => e, 150, true)(async () => {
-			await timeout(50);
-			this.checkShell();
-		});
-
-		// We want to fire a new check for the shell on a linefeed, but only
-		// when parsing has finished which is indicated by the cursormove event.
-		// If this is done on every linefeed, parsing ends up taking
-		// significantly longer due to resetting timers. Note that this is
-		// private API.
-		this._xterm.onLineFeed(() => this._newLineFeed = true);
-		this._xterm.onCursorMove(() => {
-			if (this._newLineFeed) {
-				this._onCheckShell.fire(undefined);
-				this._newLineFeed = false;
-			}
-		});
-
-		// Fire a new check for the shell when any key is pressed.
-		this._xterm.onKey(() => this._onCheckShell.fire(undefined));
+		this.checkShell();
 	}
 
-	private checkShell(): void {
+	@debounce(500)
+	async checkShell(): Promise<void> {
 		if (platform.isWindows) {
-			// TODO: Only fire when it's different
-			this.getShellName().then(title => this._onShellNameChange.fire(title));
+			await timeout(300);
```

Comment:
```suggestion
			// Wait to give the shell some time to actually launch a process, this
			// could lead to a race condition but it would be recovered from when
			// data stops and should cover the majority of cases
			await timeout(300);
```